modules/cart/commerce_cart.module

```diff
@@ -223,3 +247,19 @@ function commerce_cart_views_data_alter(array &$data) {
 function template_preprocess_commerce_cart_block(&$variables) {
   $variables['attributes']['class'][] = 'cart--cart-block';
 }
+
+/**
+ * Implements hook_cron().
+ */
+function commerce_cart_cron() {
+  $queue = \Drupal::queue('cart_expirations');
+
+  $ids = \Drupal::entityQuery('commerce_order')
+    ->condition('cart', TRUE)
+    ->execute();
+
+  // Queue ids in groups of 50.
+  for ($i = 0; $i < count($ids); $i += 50) {
```

Comment:
Actually, I have an idea which completely changes the queue worked.

1. Load all possible order types.
1. Check if the order type has cart expiration configured
1. If cart expiration is configured, query orders that have `cart => 1` and `updated < (time() - EXPIRATION_TIME`
1. Pass all of the returned order IDs into a new queue item

The result is a queue entry per order type with all the valid IDs. Perhaps we should also pass in the order type and settings, along with the IDs. So we can easily re-validate.